c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1106 @@
+    vic_mask_t int_select;
+    vic_mask_t int_enable;
+    vic_mask_t int_disable;
+    vic_mask_t soft_int_enable;
+    vic_mask_t soft_int_disable;
+    bool protection;
+    const uint _padding[3];
+    vic_interrupt_handler_t vector_address;
+    vic_interrupt_handler_t default_vector_address;
+} vic_control_t;
+
+typedef struct {
+    uint source : 5;
+    uint enable : 1;
+    uint : 26;
+} vic_vector_control_t;
+
+ASSERT_WORD_SIZED(vic_mask_t);
+ASSERT_WORD_SIZED(vic_interrupt_handler_t);
+ASSERT_WORD_SIZED(vic_vector_control_t);
+
+// ---------------------------------------------------------------------
+// 6. Counter/Timer
+
+typedef struct {
+    uint one_shot : 1;
+    uint size : 1;
+    uint pre_divide : 2;
+    uint : 1;
+    uint interrupt_enable : 1;
+    uint periodic_mode : 1;
+    uint enable : 1;
+    uint : 24;
+} timer_control_t;
+
+enum timer_pre_divide {
+    TIMER_PRE_DIVIDE_1 = 0,
+    TIMER_PRE_DIVIDE_16 = 1,
+    TIMER_PRE_DIVIDE_256 = 2
+};
+
+typedef struct {
+    uint status : 1;
+    uint : 31;
+} timer_interrupt_status_t;
+
+typedef struct {
+    uint load_value;
+    const uint current_value;
+    timer_control_t control;
+    uint interrupt_clear;
+    const timer_interrupt_status_t raw_interrupt_status;
+    const timer_interrupt_status_t masked_interrupt_status;
+    uint background_load_value;
+    uint _dummy;
+} timer_controller_t;
+
+ASSERT_WORD_SIZED(timer_control_t);
+ASSERT_WORD_SIZED(timer_interrupt_status_t);
+
+// ---------------------------------------------------------------------
+// 7. DMA Controller
+
+typedef struct {
+    uint _zeroes : 2;
+    uint length_words : 15;
+    uint : 2;
+    uint direction : 1; // 0 = write to TCM, 1 = write to SDRAM
+    uint crc : 1;
+    uint burst : 3;
+    uint width : 1; // 0 = word, 1 = double-word
+    uint privilege : 1;
+    uint transfer_id : 6;
+} dma_description_t;
+
+typedef struct {
+    uint uncommit : 1;
+    uint abort : 1;
+    uint restart : 1;
+    uint clear_done_int : 1;
+    uint clear_timeout_int : 1;
+    uint clear_write_buffer_int : 1;
+    uint : 26;
+} dma_control_t;
+
+typedef struct {
+    uint transferring : 1;
+    uint paused : 1;
+    uint queued : 1;
+    uint write_buffer_full : 1;
+    uint write_buffer_active : 1;
+    uint : 5;
+    uint transfer_done : 1;
+    uint transfer2_done : 1;
+    uint timeout : 1;
+    uint crc_error : 1;
+    uint tcm_error : 1;
+    uint axi_error : 1;
+    uint user_abort : 1;
+    uint soft_reset : 1;
+    uint : 2;
+    uint write_buffer_error : 1;
+    uint : 3;
+    uint processor_id : 8;
+} dma_status_t;
+
+typedef struct {
+    uint bridge_buffer_enable : 1;
+    uint : 9;
+    uint transfer_done_interrupt : 1;
+    uint transfer2_done_interrupt : 1;
+    uint timeout_interrupt : 1;
+    uint crc_error_interrupt : 1;
+    uint tcm_error_interrupt : 1;
+    uint axi_error_interrupt : 1;
+    uint user_abort_interrupt : 1;
+    uint soft_reset_interrupt : 1;
+    uint unknown1 : 1;
```

Comment:
how about unknown: 2? if they are truely unknown and never set.

Reply:
Gone